send emails to ros2_build_farm mailing list

mikaelarguedas opened the pull request:
With this PR, unsuccessfull jobs will now send emails to `ros2_build_farm` instead of `ros`

dirk-thomas commented:
How can individuals subscribe / unsubscribe to this list on their own?

mikaelarguedas commented:
I planned on inviting everybody on the ROS team at first and then allow anybody with an email under `osrfoundation.org` domain to unsubscribe/resubscribe. I'm looking into list configuration right now. I'll put this PR in review once configuration of the mailing list is ready.
Any comment on the strategy to manage members and subscription is welcome

dirk-thomas commented:
I would like to be able to subscribe and unsubscribe without the need for another person. Afaik that is not the case for ros@osrf. It is possible for the Google mailing lists we use for the ROS 1 farm though.

tfoote commented:
ros@osrf is a google group too: https://groups.google.com/a/osrfoundation.org/forum/#!forum/ros

mikaelarguedas commented:
the current configuration should allow you to join without the need for another  person. If any of you is willing to give it a try, I'd be happy to adjust settings if it doesn't behave the way we want it to

nuclearsandwich commented:
Stumbled upon this while working on other PRs. The "conflict" is trivial, the nightly job start time was changed in https://github.com/ros2/ci/commit/6dd1a5047d1b0ba5c66fa09bd47351d9f2603e53

A quick edit would make this mergeable again if the change is desired.

mikaelarguedas commented:
Sounds good, thanks for pointing it out!
I'll resolve once the mailing list works as expected and this is ready for merging

mikaelarguedas commented:
Rebased and updated

dirk-thomas commented:
The patch looks good.

Can you please configure the new mailing list in the same way as the existing ROS1 buildfarm mailing list.

mikaelarguedas commented:
hmm I thought that's what I did. Could you point me to the discrepancies so that I can adapt the settings ? otherwise as an owner you should be able to make the appropriate changes directly if you prefer

dirk-thomas commented:
I couldn't even see the settings as a member. Now being an owner I updated the posting access to "Not allowed" for all everyone except the buildfarm email. I also changed the delivery for the buildfarm email to "No email" (since that inbox is not checked by anyone). Currently the delivery option is set to "All email" for everyone else. Should it stay on and anybody who doesn't want to get the emails change hisdelivery setting?

mikaelarguedas commented:
>I couldn't even see the settings as a member. Now being an owner I updated the posting access to "Not allowed" for all everyone except the buildfarm email. 

Yeah I kept the posting before for testing purposes, thanks for putting it back to "Not Allowed".

> I also changed the delivery for the buildfarm email to "No email" (since that inbox is not checked by anyone). 

:+1: 

> Currently the delivery option is set to "All email" for everyone else. Should it stay on and anybody who doesn't want to get the emails change hisdelivery setting?

I figured we will let people know at buildcop selection time this afternoon how to change it and anybody will then change their own setting according to their needs.

Thanks for looking into it

mikaelarguedas commented:
Merging this. For now all emails will go to the same mailing list. I'll deploy the changes tonight before the nightlies

dirk-thomas commented:
The global email address configured in Jenkins which is e.g. used for the node watcher needs to be updated too.

mikaelarguedas commented:
Yup I'll update it at the same time this change is deployed

mikaelarguedas commented:
deployed along #41 
Every node + Jenkins Sys admin addresses updated accordingly
